respokeSDK/src/main/java/com/digium/respokesdk/RespokeClient.java

```diff
@@ -191,6 +188,29 @@
 
 
     /**
+     * A listener interface to receive a notification when the request to retrieve the history
+     * of messages for a list of groups has completed
+     */
+    public interface GroupHistoriesCompletionListener {
+
+        void onSuccess(HashMap groupsToMessages);
```

Comment:
I would specify Map here instead of HashMap. The implementation detail of the map shouldn't be part of the API.